src/app/server/RendezvousServer.cpp

```diff
@@ -36,15 +39,18 @@ using namespace ::chip::DeviceLayer;
 namespace chip {
 
 namespace {
+#if CONFIG_DEVICE_LAYER
```

Comment:
Please express the root cause of needing these new `#ifdef`. If it is that we have incorrect APIs for your usecases, then others will suffer too, and we need to converge/fix, rather than hard-to-maintain and not trivial ifdefs.

Reply:
I had to put them in because I was creating a build dependency from the controller to app/server. When I moved calls to InitServer (and then to other app/server functions to the example app main.cpp function), I did not need those #ifdefs. The latest diff does not touch any of the classes in app/server.